src/handwritten/types.py

```diff
@@ -54,3 +54,9 @@
 
 # endregion
 
+# region ID Pin namedtuples
+
+IDPinMemory = collections.namedtuple(
+    'IDPinMemory', ['data', 'format_code'])
```

Comment:
What is data, an ndarray or a list[int]?

Please use https://docs.python.org/3/library/typing.html#typing.NamedTuple for this new named tuple.

I have a branch somewhere that updates the other named tuples to be typed, I should dig that up.